hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3186,6 +3192,132 @@ public ServiceInfoEx getServiceInfo() throws IOException {
     return serviceInfo.provide();
   }
 
+  @Override
+  public ListOpenFilesResult listOpenFiles(String path,
+                                           long maxKeys,
+                                           String contToken)
+      throws IOException {
+
+//    metrics.incListOpenFiles();  // TODO: Do we want a counter for this?
+
+    final UserGroupInformation ugi = getRemoteUser();
+    if (!isAdmin(ugi)) {
+      final OMException omEx = new OMException(
+          "Only Ozone admins are allowed to list open files.",
+          PERMISSION_DENIED);
+      AUDIT.logWriteFailure(buildAuditMessageForFailure(
+          OMAction.LIST_OPEN_FILES, new LinkedHashMap<>(), omEx));
+      throw omEx;
+    }
+
+    final String dbOpenKeyPrefix;
+    final Table<String, OmKeyInfo> openKeyTable;
+    if (path == null || path.isEmpty() || path.equals(OM_KEY_PREFIX)) {
+      // path is root
+      dbOpenKeyPrefix = "";
+      // default to FSO. TODO: Add client option to pass OBS/LEGACY?
+      openKeyTable =
+          metadataManager.getOpenKeyTable(BucketLayout.FILE_SYSTEM_OPTIMIZED);
```

Comment:
I think it depends on whether or not we want to export only keys visible in the namespace. If so, then this semantics is only useful for FSO.  If not, there could be use cases where it makes sense for legacy/OBS